osu.Game.Tests/Rulesets/Scoring/ScoreProcessorTest.cs

```diff
@@ -53,5 +53,222 @@ public void TestSingleOsuHit(ScoringMode scoringMode, HitResult hitResult, int e
+        [TestCase(ScoringMode.Standardised, HitResult.Good, HitResult.Perfect, 3_350_000 / 7.0)]
+        [TestCase(ScoringMode.Standardised, HitResult.Great, HitResult.Great, 575_000)]
+        [TestCase(ScoringMode.Standardised, HitResult.Perfect, HitResult.Perfect, 575_000)]
+        [TestCase(ScoringMode.Standardised, HitResult.LargeTickMiss, HitResult.LargeTickHit, 0)]
+        [TestCase(ScoringMode.Standardised, HitResult.LargeTickHit, HitResult.LargeTickHit, 575_000)]
+        [TestCase(ScoringMode.Standardised, HitResult.SmallBonus, HitResult.SmallBonus, 30)]
+        [TestCase(ScoringMode.Standardised, HitResult.LargeBonus, HitResult.LargeBonus, 150)]
+        [TestCase(ScoringMode.Classic, HitResult.Miss, HitResult.Great, 0)]
+        [TestCase(ScoringMode.Classic, HitResult.Meh, HitResult.Great, 156)]
+        [TestCase(ScoringMode.Classic, HitResult.Ok, HitResult.Great, 312)]
+        [TestCase(ScoringMode.Classic, HitResult.Good, HitResult.Perfect, 3744 / 7.0)]
+        [TestCase(ScoringMode.Classic, HitResult.Great, HitResult.Great, 936)]
+        [TestCase(ScoringMode.Classic, HitResult.Perfect, HitResult.Perfect, 936)]
+        [TestCase(ScoringMode.Classic, HitResult.LargeTickMiss, HitResult.LargeTickHit, 0)]
+        [TestCase(ScoringMode.Classic, HitResult.LargeTickHit, HitResult.LargeTickHit, 936)]
+        [TestCase(ScoringMode.Classic, HitResult.SmallBonus, HitResult.SmallBonus, 30)]
+        [TestCase(ScoringMode.Classic, HitResult.LargeBonus, HitResult.LargeBonus, 150)]
+        public void TestFourVariousResultsOneMiss(ScoringMode scoringMode, HitResult hitResult, HitResult maxResult, double expectedScore)
+        {
+            var minResult = new TestJudgement(hitResult).MinResult;
+
+            IBeatmap fourObjectBeatmap = new TestBeatmap(new RulesetInfo())
+            {
+                HitObjects = new List<HitObject>(Enumerable.Repeat(new TestHitObject(maxResult), 4))
+            };
+            scoreProcessor.Mode.Value = scoringMode;
+            scoreProcessor.ApplyBeatmap(fourObjectBeatmap);
+
+            for (int i = 0; i < 4; i++)
+            {
+                var judgementResult = new JudgementResult(fourObjectBeatmap.HitObjects[i], new Judgement())
+                {
+                    Type = i == 2 ? minResult : hitResult
+                };
+                scoreProcessor.ApplyResult(judgementResult);
+            }
+
+            Assert.IsTrue(Precision.AlmostEquals(expectedScore, scoreProcessor.TotalScore.Value));
+        }
+
+        [TestCase(ScoringMode.Standardised, HitResult.SmallTickHit, 6_850_000 / 7.0)]
+        [TestCase(ScoringMode.Standardised, HitResult.SmallTickMiss, 6_400_000 / 7.0)]
+        [TestCase(ScoringMode.Classic, HitResult.SmallTickHit, 1950 / 7.0)]
+        [TestCase(ScoringMode.Classic, HitResult.SmallTickMiss, 1500 / 7.0)]
+        public void TestSmallTicksAccuracy(ScoringMode scoringMode, HitResult hitResult, double expectedScore)
```

Comment:
i don't think fixing the issue with the old test by swapping it for a different test is a proper resolution. it's skirting the problem. i think we should fix the scoring issue instead as it's clearly unintuitive behaviour.

however, to confirm for sure - @peppy, @smoogipoo, what are your views on this?

Reply:
So I guess there's really two issues here - the first being that `Classic` scoring is forced which shouldn't be required anymore (see my other comment above). The second is that classic scoring gives 0 score if there's 0 combo, which should definitely not be the case.

https://github.com/ppy/osu/blob/f206ea96ce6ccbce87af0e360b0f2630ceac0635/osu.Game/Rulesets/Scoring/ScoreProcessor.cs#L224-L226
This should probably be something like `(accuracyRatio * (maxCombo + 1) * 300)`

Reply:
But should standard scoring award combo portion score if the combo is 0 achieved out of 0 max combo?

Reply:
Or it might make sense to have accuracy portion be 100% if max achievable combo is 0

Reply:
No. As it is is intended - combo should be fully awarded if max combo is 0.

Scoring should have zero edge cases that require special casing.

Reply:
I changed comboRatio from being 0 if maxHighestCombo is 0 to being 1. Then all the combo portion score is awarded when the map has 0 combo FC